archive/c/clojure/bubble-sort.clj

```diff
@@ -0,0 +1,17 @@
+(ns bubble-sort
+	(:gen-class))
+  
+(defn- bubble [ys x]
+  (if-let [y (peek ys)]
+    (if (> y x)
+      (conj (pop ys) x y)
+      (conj ys x))
+    [x]))
+
+(defn bubble-sort [xs]
+  (let [ys (reduce bubble [] xs)]
+    (if (= xs ys)
+      xs
+      (recur ys))))
+
+(bubble-sort [3 2 4 1]) ; [1 2 3 4]
```

Comment:
Is there any way to pull this array from the command line?

Reply:
Not directly. It can be printed in the same way like in the hello-world.clj, if you like to.

Reply:
That's fine! That's all I needed to know.